src/view_maybe/view_maybe.h

```diff
@@ -8,130 +8,91 @@
 #include <stl2/view/view_interface.hpp>
 
 
-template<class> struct dereference_type {};
-
-template<class D>
-requires
-requires(const D& d) {{ *d } -> auto&&; }
-struct dereference_type<D> {
-    using type = decltype(*std::declval<const D&>());
-};
-
-template<class D>
-using dereference_t = typename dereference_type<D>::type;
-
-
-template<class T>
-concept bool ContextualBool =
-    requires(const T& t) {
-    {bool(t)} -> bool;
-};
+template <class T, class Ref, class ConstRef>
+concept bool _Nullable2 =
+    std::is_lvalue_reference_v<Ref> &&
+    std::is_object_v<std::remove_reference_t<Ref>> &&
+    std::is_lvalue_reference_v<ConstRef> &&
+    std::is_object_v<std::remove_reference_t<ConstRef>> &&
+    ranges::ConvertibleTo<std::add_pointer_t<ConstRef>, 
+        const std::remove_reference_t<Ref>*>;
 
 template <class T>
 concept bool Nullable =
-    std::experimental::ranges::Readable<std::remove_reference_t<T>> &&
-    ContextualBool<T> &&
-    requires (const T& t) {
-    typename dereference_t<T>;
-    std::is_object_v<dereference_t<T>>;
-};
-
+    std::is_object_v<T> &&
+    requires(T& t, const T& ct) {
+        bool(ct);
+        *t;
+        *ct;
+    } &&
+    _Nullable2<T, ranges::iter_reference_t<T>, ranges::iter_reference_t<const T>>;
 
 template <Nullable Maybe>
```

Comment:
I missed a constraint; `semiregular_box` (*`semiregular`* in the working draft) needs its type parameter to model `CopyConstructible` in order to model `Semiregular` so `safe_maybe_view` can model `View`:
```suggestion
 template <Nullable Maybe>
     requires ranges::CopyConstructible<Maybe>
```